sdk/src/azure/iot/az_iot_hub_client_telemetry.c

```diff
@@ -33,13 +36,27 @@ AZ_NODISCARD az_result az_iot_hub_client_telemetry_get_publish_topic(
   int32_t required_length = az_span_size(telemetry_topic_prefix)
       + az_span_size(client->_internal.device_id) + az_span_size(telemetry_topic_suffix);
   int32_t module_id_length = az_span_size(*module_id);
+  int32_t content_type_length = az_span_size(client->_internal.options.content_type);
+  int32_t content_encoding_length = az_span_size(client->_internal.options.content_encoding);
   if (module_id_length > 0)
   {
     required_length += az_span_size(telemetry_topic_modules_mid) + module_id_length;
   }
+  if (content_type_length > 0)
+  {
+    required_length += az_span_size(telemetry_content_type) + content_type_length;
+  }
+  if (content_encoding_length > 0)
+  {
+    required_length += az_span_size(telemetry_content_encoding) + content_encoding_length
+        + (content_type_length > 0 ? (int32_t)sizeof(telemetry_param_separator) : 0);
```

Comment:
nit: Is there much benefit from using `sizeof(...)` here for a single uint8_t? Seems overly complicated (needing a cast too). If we were to change the separator for some reason, it will likely change the data type and would require us to update this anyway. Maybe use a comment instead?
```suggestion
    required_length += az_span_size(telemetry_content_encoding) + content_encoding_length
        + (content_type_length > 0 ? 1 : 0); // separator is single byte, '&'
```